glean-core/src/debug.rs

```diff
@@ -147,23 +191,49 @@ pub fn validate_debug_view_tag(value: String) -> Option<String> {
     loop {
         match iter.next() {
             // We are done, so the whole expression is valid.
-            None => return Some(value),
+            None => return true,
             // Valid characters.
             Some('-') | Some('a'..='z') | Some('A'..='Z') | Some('0'..='9') => (),
             // An invalid character
             Some(c) => {
-                log::error!("Invalid character '{}' in debug view tag.", c);
-                return None;
+                log::error!("Invalid character '{}' in the tag.", c);
+                return false;
             }
         }
         count += 1;
         if count == 20 {
-            log::error!("Debug view tag cannot exceed 20 characters");
-            return None;
+            log::error!("A tag cannot exceed 20 characters.");
+            return false;
         }
     }
 }
 
+/// Validate the list of source tags.
+///
+/// This builds upon the existing `validate_tag` function, since all the
+/// tags should respect the same rules to make the pipeline happy.
+fn validate_source_tags(tags: Vec<String>) -> bool {
+    if tags.is_empty() {
+        return false;
+    }
+
+    if tags.len() > GLEAN_MAX_SOURCE_TAGS {
+        log::error!(
+            "A list of tags cannot contain more than {} elements.",
+            GLEAN_MAX_SOURCE_TAGS
+        );
+        return false;
+    }
+
+    for tag in tags {
```

Comment:
Unlike `tokenize_string`, I don't think this rejects tags starting with `glean`.  We should be consistent (and also log / return error).